lib/archethic/contracts/contract.ex

```diff
@@ -94,4 +97,29 @@ defmodule Archethic.Contracts.Contract do
       ) do
     Map.update!(contract, :conditions, &Map.put(&1, condition_name, conditions))
   end
+
+  @doc """
+  Add a public or private function to the contract
+  """
+  def add_function(
```

Comment:
Using the visibility inside function attributes allow to have only one function `add_function` with the visibility in parameter